devtools/x/src/config.rs

```diff
@@ -25,7 +25,38 @@ pub struct Config {
     fix: Fix,
     /// Cargo configuration
     cargo: CargoConfig,
-    tools: Vec<(String, String)>,
+    grcov: CargoTool,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct CargoTool {
+    pub installer: CargoInstallation,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct Sccache {
+    /// Sccache Url
+    pub installer: CargoInstallation,
+    pub bucket: String,
+    pub prefix: Option<String>,
+    /// utility of this seems to change in stable/vs rusoto, left for completeness.
+    pub endpoint: Option<String>,
+    pub region: Option<String>,
+    ///only used in stable, delete when rusoto is merged in
+    pub ssl: Option<bool>,
+    pub public: Option<bool>,
+    pub envs: Option<Vec<(String, String)>>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct CargoInstallation {
+    pub version: String,
+    pub git: Option<String>,
+    pub git_rev: Option<String>,
+    pub features: Option<Vec<String>>,
```

Comment:
Some documentation on what this is trying to describe would be useful. It sort of seems like the version and git parts are mutually exclusive (given how its used below) perhaps this data structure should be better defined to describe that (ie using an enum)

Reply:
Unfortunately the git revision and the version aren't quite mutually exclusive. I have to check the version to know whether to update or not.  Ideally pre release versions would contain their git hash.   But that's probably a harder sell to the authors of projects.